src/server/pfs/drive/driver.go

```diff
@@ -1071,6 +1072,11 @@ func (d *driver) inspectFile(file *pfs.File, filterShard *pfs.Shard, shard uint6
 	if fileInfo.FileType == pfs.FileType_FILE_TYPE_NONE {
 		return nil, nil, pfsserver.NewErrFileNotFound(file.Path, file.Commit.Repo.Name, file.Commit.ID)
 	}
+	// We return NotFound if it's a regular file but its blocks do not match the
+	// given filter shard
+	if fileInfo.FileType == pfs.FileType_FILE_TYPE_REGULAR && len(blockRefs) == 0 && (filterShard != nil && filterShard.BlockNumber > 1) {
+		return nil, nil, pfsserver.NewErrFileNotFound(file.Path, file.Commit.Repo.Name, file.Commit.ID)
+	}
```

Comment:
That's not what I'm asking.

It seems like there are 5 states we can be in when we hit this line:

- File had blocks, we didn't have a filterShard -> FileFound
- File had blocks, we did have a filterShard but some blocks matched it -> FileFound
- File had blocks, we did have a filterShard but it filtered all the blocks -> FileNotFound
- File had no blocks, we didn't have a filterShard -> FileFound
- File had no block, we did have a filterShard -> FileNotFound

In the cases where the File has blocks checking the filterShard is redundant because the only way a file can wind up with `FILE_TYPE_REGULAR` is if some of the blocks get filtered, which implies that it's not nil.

And in the case where the file truly has no blocks, which actually isn't possible right now I believe, checking the filterShard doesn't seem to do anything sensible. Adding a `filterShard` somehow takes a `inspect-file` from `Found` to `NotFound` even though it doesn't actually filter any blocks. Which seems like a really weird behavior to me.